src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy

```diff
@@ -1020,6 +1030,227 @@ class NutanixPrismElementPluginProvisionProvider extends AbstractProvisionProvid
+			context.async.storageVolume.remove([storageVolume], computeServer, true).blockingGet()
+		}
+
+		// This operation is considered failed if any of the request's resizes failed.
+		rtn.success = !rtn.error
+		return rtn
+	}
+
+	ServiceResponse resizeNetworks(HttpApiClient client, ComputeServer server, ResizeRequest resizeRequest) {
+		// existing code does nothing for network updates so we don't either
+		if(!resizeRequest.interfacesAdd && !resizeRequest.interfacesDelete) {
+			return ServiceResponse.success()
+		}
+		def rtn = ServiceResponse.prepare()
+		def vmId = server.externalId
+		def existingNics = NutanixPrismElementApiService.getVirtualMachineNics(client, server.cloud, vmId)?.nics
+		def computeServer = context.services.computeServer.get(server.id)
+		resizeRequest.interfacesAdd?.each { newInterfaceOpts ->
+			log.info("adding network: ${newInterfaceOpts}")
+			def targetNetwork = context.services.network.get(newInterfaceOpts.network.id.toLong())
+			if (!targetNetwork) {
+				log.error("couldn't find network id: ${newInterfaceOpts.network.id}. skipping...")
+				return
+			}
+
+			def networkConfig = [
+				zone:server.cloud,
+				networkUuid:targetNetwork.uniqueId,
+				ipAddress:newInterfaceOpts.ipAddress
+			]
+			def networkResults = NutanixPrismElementApiService.addNic(client, networkConfig, server.externalId)
+			log.info("network results: ${networkResults}")
+			if(networkResults.success == true) {
+				def newInterface = NutanixPrismElementSyncUtility.buildComputeServerInterface(context, computeServer, newInterfaceOpts.network)
+				def vmNics = NutanixPrismElementApiService.getVirtualMachineNics(client, server.cloud, server.externalId)?.nics
+				vmNics.each { vmNic ->
+					def existingNic = existingNics.find{it.macAddress == vmNic.macAddress}
+					if(!existingNic) {
+						newInterface.externalId = vmNic.macAddress
+						newInterface.publicIpAddress = vmNic.requestedIpAddress
+					}
+				}
+				newInterface.uniqueId = "morpheus-nic-${computeServer.id}-${computeServer.interfaces.size()}"
+				context.services.computeServer.computeServerInterface.create(newInterface)
+				computeServer.interfaces.add(newInterface)
+			}
+		}
+		resizeRequest.interfacesDelete?.eachWithIndex { networkDelete, index ->
+			def deleteConfig = [zone:server.cloud, macAddress: networkDelete.macAddress]
+			def deleteResults = NutanixPrismElementApiService.deleteNic(client, deleteConfig, server.externalId, networkDelete.externalId)
+			log.debug("deleteResults: ${deleteResults}")
+			if(deleteResults.success == true) {
+				def networkInterface = context.services.computeServer.computeServerInterface.get(networkDelete.id)
+				computeServer.interfaces.remove(networkInterface)
+				// TODO: switch back to bulkRemove once fixed
+				context.async.computeServer.computeServerInterface.remove([networkDelete], computeServer).blockingGet()
+			}
+		}
+
+		context.services.computeServer.save(computeServer)
+		return rtn
```

Comment:
Same thing here. Should this propagate a failure up? If not, can we add a comment?

Reply:
will add comment